filesystems/general/pjd-fstest/runtest.sh

```diff
@@ -0,0 +1,254 @@
+			report_result $TEST WARN
+		# Abort the task
+		rstrnt-abort --server $RSTRNT_RECIPE_URL/tasks/$TASKID/status
+		exit 0
+		;;
+	esac
+	return 0
+}
+
+
+build_pjd_fstest()
+{
+	local workdir=$1
+	pushd $workdir
+	rlRun -l "git clone git://git.code.sf.net/p/ntfs-3g/pjd-fstest" "0-255" "Cloning pjd-fstest repo"
+	if [ $? -ne 0 ]; then
+		echo "WARN : Failed cloning pjd-fstest" | tee -a $OUTPUTFILE
+		report_result $TEST WARN
+		# Abort the task
+		rstrnt-abort --server $RSTRNT_RECIPE_URL/tasks/$TASKID/status
+		exit 0
+	fi
+	rlRun "pushd pjd-fstest"
+	rlRun "make" "0-255" "Building test suite"
+	if [ $? -ne  0 ] ; then
+		echo "WARN : Failed compiling pjd-fstest" | tee -a $OUTPUTFILE
+		report_result $TEST WARN
+		# Abort the task
+		rstrnt-abort --server $RSTRNT_RECIPE_URL/tasks/$TASKID/status
+		exit 0
+	fi
+	rlRun "popd"
+	rlRun "TestsDir=\`ls -d $workdir/pjd-fstest/tests\`" 0 "Looking for tests"
+	rlAssertExists "$TestsDir"
+}
+
+
+pjd_fstest_prepare() {
+	rlPhaseStartSetup ${TEST}:Build
+		rlRun "TmpDir=\`mktemp -d\`" 0 "Creating tmp directory"
+		build_pjd_fstest $TmpDir
+	rlPhaseEnd
+}
+
+localfs_prepare()
+{
+	rlPhaseStartSetup ${TEST}:Setup:$FSTYPE
+                # prepare for local fstest
+		rlRun "echo -e \"os=Linux\nfs=${FSTYPE}\" > $TestsDir/conf" 0 "Setting up environment"
+
+		# we have a pre-mount test partition, grab it
+		if test -n "$TEST_MNT"; then
+			rlRun "TEST_DEV=`cat /proc/mounts | grep "$TEST_MNT" | awk '{print $1}'`"
+			# Remove entry in fstab
+			rlRun "cp /etc/fstab{,.bak}"
+			rlRun "cat /etc/fstab.bak | grep -v $TEST_MNT > /etc/fstab"
+			rlRun "umount $TEST_DEV" 0 "umount device for re-creating filesystem"
+		fi
+
+		if test -z "${TEST_DEV}"; then
+			# Check for a pre-prepared lvm device
+			TMP_DEV="$(echo /dev/mapper/*-xfstest)"
+			echo "TMP_DEV=$TMP_DEV"
+			if test -b "$TMP_DEV";then
+				TEST_DEV="$TMP_DEV"
+				dd if=/dev/zero of="$TEST_DEV" bs=1M count=1
+			fi
+		fi
+
+		if test -z "${TEST_DEV}"; then
+			if [ $RHEL_DISTRO -ge 5 ] || rlIsFedora ;then
+				TEST_DEV=`losetup -f`
+			else
+				i=0
+				# rhel 4 doesn't support losetup -f
+				while true; do
+					if [ ! -e /dev/loop${i} ];then
+						TEST_DEV=/dev/loop${i}
+						break
+					fi
+					let i++
+				done
+			fi
+			TEST_DEV_LOOP=1
+			rlRun "dd if=/dev/zero of=test.img bs=1M count=500"
+			rlRun "losetup $TEST_DEV test.img"
+		fi
+
+		rpm -q xfsprogs
+		if [ $? -ne 0 -a $FSTYPE = "xfs" ]; then
+			${yum} -y install xfsprogs
+			if [ $? -ne 0 ]; then
+				echo "WARN : Failed installing xfsprogs rpm package" | tee -a $OUTPUTFILE
+				report_result $TEST WARN
+				# Abort the task
+				rstrnt-abort --server $RSTRNT_RECIPE_URL/tasks/$TASKID/status
+				exit 0
+			fi
+		fi
+
+		if [[ $FSTYPE =~ ext ]] ; then
+			rlRun "/sbin/mkfs -t $FSTYPE -F $TEST_DEV $MKFS_OPTS" 0 "Creating filesystem on image"
+		else
+			rlRun "/sbin/mkfs -t $FSTYPE -f $TEST_DEV $MKFS_OPTS" 0 "Creating filesystem on image"
+		fi
```

Comment:
I wonder if it would be better that this whole `if` only had the flag (calculating if it's `-f` or `-F`) and then `rlRun` would be outside, using the flag calculated in the `if`...

Reply:
this is copied from original code, we could  change.  Trying to preserve the original code :-)